unix_integration/src/daemon.rs

```diff
@@ -193,6 +195,7 @@ async fn handle_client(
     };
 
     let mut reqs = Framed::new(sock, ClientCodec);
+    let mut pam_state = PamState::Uninitialized;
```

Comment:
The pam state machine only has 2 states here; Uninitialized or Step (meaning it's performing auth step operations). While the id provider returns PamPrompt requests, the machine will remain in the Step state (which holds the auth id). As soon as any other response is returned (either Success or failure, etc) from `pam_account_authenticate_step`, the state returns to Uninitialized.